app/src/org/commcare/network/HttpRequestGenerator.java

```diff
@@ -197,6 +197,10 @@ public HttpResponse makeKeyFetchRequest(String baseUri, Date lastRequest) throws
             url = url.buildUpon().appendQueryParameter("last_issued", DateUtils.formatTime(lastRequest, DateUtils.FORMAT_ISO8601)).build();
         }
 
+        // include IMEI in key fetch request for auditing large deployments
+        url = url.buildUpon().appendQueryParameter("device_id",
+                CommCareApplication.instance().getPhoneId()).build();

```

Comment:
I wonder if we should just use the current approach of adding the device id to requests: https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/network/HttpRequestGenerator.java#L205-L212